File: LC_Gen_Codes/2917_no_setting.py
class Solution:
    def findKOr(self, nums: List[int], k: int) -> int:
        nums_len = len(nums)
        result = 0
        
        for i in range(32):  # considering 32 bits in an integer
            count = 0
            
            for num in nums:
                if num & (1 << i) != 0:
                    count += 1
                    
                if count >= k:
                    result |= (1 << i)
                    break
        
        return result

File: LC_Gen_Codes_Java/2917_no_setting.java
java
class Solution {
    public int findKOr(int[] nums, int k) {
        int numsLen = nums.length;
        int result = 0;
        
        for (int i = 0; i < 32; i++) {  // considering 32 bits in an integer
            int count = 0;
            
            for (int num : nums) {
                if ((num & (1 << i)) != 0) {
                    count++;
                }
                
                if (count >= k) {
                    result |= (1 << i);
                    break;
                }
            }
        }
        
        return result;
    }
}